VS2019 16.7.0 build failure

I encounter new problem building msquic.kernel.sln possibly after I upgrade VS2019 to 16.7.0.

msquic\src\platform\datapath_winkernel.c(2202): error C6011: Dereferencing NULL pointer 'Binding'. 
msquic\src\platform\hashtable.c(731): error C6387: 'ContextPtr->ChainHead' could be '0':  this does not adhere to the specification for the function 'QuicListIsEmpty'.

--- a/src/core/connection.c
+++ b/src/core/connection.c
@@ -3265,6 +3265,7 @@ QuicConnRecvHeader(
 
             QUIC_DBG_ASSERT(TokenBuffer == NULL);
             QuicPacketDecodeRetryTokenV1(Packet, &TokenBuffer, &TokenLength);
+            QUIC_DBG_ASSERT(TokenBuffer != NULL);
             QUIC_DBG_ASSERT(TokenLength == sizeof(QUIC_RETRY_TOKEN_CONTENTS));
 
             QUIC_RETRY_TOKEN_CONTENTS Token;

--- a/src/core/listener.c
+++ b/src/core/listener.c
@@ -125,12 +125,13 @@ MsQuicListenerClose(
     MsQuicListenerStop(Handle);
 
     QuicRundownUninitialize(&Listener->Rundown);
-    QUIC_FREE(Listener);
 
     QuicTraceEvent(
         ListenerDestroyed,
         "[list][%p] Destroyed",
         Listener);
+
+    QUIC_FREE(Listener);
     QuicRundownRelease(&Session->Rundown);
 
     QuicTraceEvent(

--- a/src/core/lookup.c
+++ b/src/core/lookup.c
@@ -55,7 +55,10 @@ QuicLookupUninitialize(
         for (uint8_t i = 0; i < Lookup->PartitionCount; i++) {
             QUIC_PARTITIONED_HASHTABLE* Table = &Lookup->HASH.Tables[i];
             QUIC_DBG_ASSERT(Table->Table.NumEntries == 0);
+#pragma warning(push)
+#pragma warning(disable:6001)
             QuicHashtableUninitialize(&Table->Table);
+#pragma warning(pop)
             QuicDispatchRwLockUninitialize(&Table->RwLock);
         }
         QUIC_FREE(Lookup->HASH.Tables);
@@ -199,7 +202,10 @@ QuicLookupRebalance(
             for (uint8_t i = 0; i < PreviousPartitionCount; i++) {
                 QUIC_HASHTABLE_ENTRY* Entry;
                 QUIC_HASHTABLE_ENUMERATOR Enumerator;
+#pragma warning(push)
+#pragma warning(disable:6001)
                 QuicHashtableEnumerateBegin(&PreviousTable[i].Table, &Enumerator);
+#pragma warning(pop)
                 while (TRUE) {
                     Entry = QuicHashtableEnumerateNext(&PreviousTable[i].Table, &Enumerator);
                     if (Entry == NULL) {
@@ -219,7 +225,10 @@ QuicLookupRebalance(
                         CID,
                         FALSE);
                 }
+#pragma warning(push)
+#pragma warning(disable:6001)
                 QuicHashtableUninitialize(&PreviousTable[i].Table);
+#pragma warning(pop)
             }
             QUIC_FREE(PreviousTable);
         }

--- a/src/core/sent_packet_metadata.c
+++ b/src/core/sent_packet_metadata.c
@@ -29,6 +29,8 @@ QuicSentPacketMetadataReleaseFrames(
     for (uint8_t i = 0; i < Metadata->FrameCount; i++) {
         switch (Metadata->Frames[i].Type)
         {
+#pragma warning(push)
+#pragma warning(disable:6001)
         case QUIC_FRAME_RESET_STREAM:
             QuicStreamRelease(Metadata->Frames[i].RESET_STREAM.Stream, QUIC_STREAM_REF_SEND_PACKET);
             break;
@@ -44,6 +46,7 @@ QuicSentPacketMetadataReleaseFrames(
         case QUIC_FRAME_STREAM:
             QuicStreamRelease(Metadata->Frames[i].STREAM.Stream, QUIC_STREAM_REF_SEND_PACKET);
             break;
+#pragma warning(pop)
         default:
             //
             // Nothing to clean up for other frame types.

--- a/src/platform/datapath_winkernel.c
+++ b/src/platform/datapath_winkernel.c
@@ -2195,7 +2195,7 @@ QuicDataPathBindingReturnRecvDatagrams(
         QuicPoolFree(BatchedInternalContext->OwningPool, BatchedInternalContext);
     }
 
-    if (DataIndications != NULL && Binding != NULL) {
+    if (DataIndications != NULL) {
         //
         // Return the datagram indications back to Wsk.
         //

--- a/src/platform/datapath_winkernel.c
+++ b/src/platform/datapath_winkernel.c
@@ -2195,10 +2195,11 @@ QuicDataPathBindingReturnRecvDatagrams(
         QuicPoolFree(BatchedInternalContext->OwningPool, BatchedInternalContext);
     }
 
-    if (DataIndications != NULL) {
+    if (DataIndications != NULL && Binding != NULL) {
         //
         // Return the datagram indications back to Wsk.
         //
+        QUIC_DBG_ASSERT(Binding != NULL);
         Binding->DgrmSocket->Dispatch->WskRelease(Binding->Socket, DataIndications);
     }
 }

--- a/src/platform/hashtable.c
+++ b/src/platform/hashtable.c
@@ -728,6 +728,8 @@ Arguments:
         ContextPtr = Context;
     }
 
+    QUIC_DBG_ASSERT(ContextPtr->ChainHead != NULL);
+
     if (QuicListIsEmpty(ContextPtr->ChainHead)) {
         HashTable->NonEmptyBuckets++;
     }

